File: languages/js/src/entity/identity_v2/decimal_enum_attribute_descriptor.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities_v2";
import { AttributeDescriptor } from "./attribute_descriptor";

/**
 * Represents a descriptor for an attribute with a decimal enum value.
 */
export class DecimalEnumAttributeDescriptor extends AttributeDescriptor {
  enumeration: number[];

  /**
   * Constructs an DecimalEnumAttributeDescriptor object with the specified parameters.
   * @param displayName 
   * @param technicalName 
   * @param description 
   * @param required 
   * @param enumeration 
   */
  constructor(
    displayName: string,
    technicalName: string,
    description: string,
    required: boolean,
    enumeration: number[]
  ) {
    super(displayName, technicalName, description, required);
    this.enumeration = enumeration;
  }

  public toProto(): identityEntitiesProto.DecimalEnumAttributeDefinitionV2 {
    return identityEntitiesProto.DecimalEnumAttributeDefinitionV2.fromPartial({
      displayName: this.displayName,
      id: this.technicalName,
      description: this.description,
      required: this.required,
      enum: this.enumeration
    });
  }

  static fromProto(
    r: identityEntitiesProto.DecimalEnumAttributeDefinitionV2
  ): DecimalEnumAttributeDescriptor {
    return new DecimalEnumAttributeDescriptor(
      r.displayName,
      r.id,
      r.description,
      r.required,
      r.enum
    );
  }
}

File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/DecimalEnumAttributeDescriptor.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.bridge.proto.IdentityEntitiesV2;
import java.util.List;

/**
 * Represents a descriptor for an attribute with a decimal enum value.
 */
public class DecimalEnumAttributeDescriptor extends AttributeDescriptor {
  private List<Double> enumeration;

  /**
   * Constructs an DecimalEnumAttributeDescriptor object with the specified parameters.
   * @param displayName
   * @param technicalName
   * @param description
   * @param required
   * @param enumeration
   */
  public DecimalEnumAttributeDescriptor(
      String displayName,
      String technicalName,
      String description,
      Boolean required,
      List<Double> enumeration) {
    super(displayName, technicalName, description, required);
    this.enumeration = enumeration;
  }

  public static DecimalEnumAttributeDescriptor fromProto(
      IdentityEntitiesV2.DecimalEnumAttributeDefinitionV2 res) {
    return new DecimalEnumAttributeDescriptor(
        res.getDisplayName(),
        res.getId(),
        res.getDescription(),
        res.getRequired(),
        res.getEnumList());
  }

  public IdentityEntitiesV2.DecimalEnumAttributeDefinitionV2 toProto() {
    return IdentityEntitiesV2.DecimalEnumAttributeDefinitionV2.newBuilder()
        .setDisplayName(this.displayName)
        .setId(this.technicalName)
        .setDescription(this.description)
        .setRequired(this.required)
        .addAllEnum(this.enumeration)
        .build();
  }
}
